### ISSUE TITLE
Problems with the translation of the theme with poedit

### ISSUE THREAD

**Issue by Pavel-Mukha** [themosis/themosis#276]
Hello, I want to understand how to translate the theme using poedit.
As it seems to me, I performed all the steps correctly according to this article - https://framework.themosis.com/docs/2.0/blade/
But the translation is not working yet.
In the template I use the following code:
`<h2>{{__('test', THEME_TD)}}</h2>`

I configured the poedit program according to the instructions.
http://joxi.ru/E2pZxPT7GZR4rY ( <--- here screenshot )

Generated my language file `ru_RU.po` using the `poedit` program - http://joxi.ru/vAW8o7igOWJZrW

And also pointed the paths for folders - http://joxi.ru/5md96bt38XbD21

Can you tell me where I made a mistake?

**Comment by jlambe** [comment]
Is the current locale of WordPress set to `ru_RU` as well ?

Also, can you try to reset views cache on the application by running the following command at application root:

```bash
php console view:clear
```

As I can see, everything seems correct. If it doesn't work, please provide me more details about your installation and its current configuration and we'll try to figure this out.

**Comment by jlambe** [comment]
I've just tested with the `ru_RU` locale and everything seems to work. Here are my steps using PoEdit:

1. Open PoEdit and choose create a new translation based on a POT file
2. Select the `themosis.pot` file located into the theme languages directory
3. Set the translation to `Russian (Russian)`
4. Edit the catalog and add the `views` directory and `inc` one
5. Save the new ru_RU.po file
6. Run a scan to find all strings to translate (view screenshot below)
7. Set the string translations
8. Save the updated file (it generates a new .mo file)
9. Login to WordPress and set the main language to russian.

Here are screenshots using a fresh installation and a new theme:

<img width="413" alt="Capture d’écran 2019-03-24 à 10 44 33" src="https://user-images.githubusercontent.com/1917490/54877688-061e1480-4e22-11e9-817b-92a73783af33.png">
<img width="170" alt="Capture d’écran 2019-03-24 à 10 44 42" src="https://user-images.githubusercontent.com/1917490/54877689-061e1480-4e22-11e9-8ffe-6173dacbd87a.png">
<img width="1792" alt="Capture d’écran 2019-03-24 à 10 45 17" src="https://user-images.githubusercontent.com/1917490/54877690-06b6ab00-4e22-11e9-9e67-c28a60b42c0d.png">

**Comment by Pavel-Mukha** [comment]
Julien thank you for the quick response.
I tried all the options, but the strings are still not translated.

> Edit the catalog and add the views directory and inc one

Which views directory should be added? The one that is in the root of the project or in the folder with the theme?
You can give me these two files, ru_RU.po and ru_RU.mo, I will copy into my project and test it.

>  If it doesn't work, please provide me more details about your installation and its current configuration and we'll try to figure this out.

Version of themosis 2.0-RC1
Poedit 1.8.7.1 (for linux)

I think there is no problem with the poedit program, because everything works on standard cms wordpress.
I think the problem is due to the fact that I do not correctly configure the file `ru_RU.po`
And most likely not associated with the blade template because even if I try to print a string using a hook, for example,` init`, the situation is the same.

```
add_action('init', fnction(){
   echo __('test', THEME_TD);
});

```
At the output, I get a string without translating `test`

**Comment by Pavel-Mukha** [comment]
Я снял пятиминутный видеоролик, чтобы показать свои действия. К сожалению интерфейс программы POEDIT установлен русский, я не смог поменять на английски. Но думаю интуитивно вам будет понятно, где и что находится в интерфейсе программы. 

https://www.youtube.com/watch?v=Q3Z6wWcIY1U

**Comment by Pavel-Mukha** [comment]
I shot a five-minute video to show my actions. Unfortunately, the interface of the POEDIT program is Russian, I could not change it to English. But I think that intuitively you will understand where and what is in the program interface.

https://www.youtube.com/watch?v=Q3Z6wWcIY1U

**Comment by jlambe** [comment]
Here are the files I used:

[translations.zip](https://github.com/themosis/themosis/files/3006784/translations.zip)

Looking at the video, you defined your string to translate inside a view from the application root. Can you try inside a theme view instead ?

I'm not sure how gettext translations is working, but I think it uses the path defined with the catalogue to recognize strings in your files. As your catalog path do not reference the views directory of the application root, I think it should not work.

Views stored inside the application root should only be used for shared views ->  mainly for administration use for example.

If you need to translate views from the root, there is an `APP_TD` constant to use and translation files should be inside the `resources/languages/$locale/$domain.mo` where `$locale` is for example `ru_RU` and `$domain` is the `APP_TD` environment variable defined inside the `.env` file.

- https://github.com/themosis/themosis/blob/master/.env.sample#L4
- https://github.com/themosis/themosis/blob/master/app/Hooks/Application.php#L41

Application structure information:
- https://framework.themosis.com/docs/2.0/structure/

**Comment by jlambe** [comment]
@Pavel-Mukha Another user found a tiny issue related to translations, perhaps this affects your installation. When installing a new theme, the `Domain Path` header was removed from the `style.css` file leading to a sort of not found directory for the theme translations.

There is a fix coming for release `2.0.1` for this but if you want to try, add the `Domain Path: languages` header to your theme `style.css` file.
